docs/source/routing/performance/caching/response-caching/customization.mdx

```diff
@@ -0,0 +1,343 @@
+        #highlight-start
+        username: root
+        password: ${env.REDIS_PASSWORD}
+        tls:
+          certificate_authorities: ${file./path/to/ca.crt}
+          client_authentication:
+            certificate_chain: ${file./path/to/certificate_chain.pem}
+            key: ${file./path/to/key.pem}
+        #highlight-end
+```
+
+### Timeout configuration
+
+Redis connections and commands have a default timeout of 500ms that you can override. A timeout of 750ms provides a good balance between responsiveness and reliability for most use cases:
+
+```yaml title="router.yaml"
+preview_response_cache:
+  enabled: true
+  subgraph:
+    all:
+      enabled: true
+      redis:
+        urls: ["redis://..."]
+        fetch_timeout: 750ms
+        insert_timeout: 750ms
+        invalidate_timeout: 750ms
+```
+
+### TTL for Redis entries
+
+The `ttl` option defines the default global expiration for Redis entries. By default it doesn't set any expiration for response caching.
+
+To prevent potential cache overflow, consider setting the TTL to 24 hours or twice the median publish interval (whichever is less), and monitor cache utilization in your environment, especially if you cache a lot of different data:
+
+```yaml title="router.yaml"
+preview_response_cache:
+  enabled: true
+  subgraph:
+    all:
+      enabled: true
+      redis:
+        urls: ["redis://..."]
+        ttl: 24h
+```
+
+### Namespace prefix
+
+When using the same Redis instance for multiple purposes, the `namespace` option defines a prefix for all the keys defined by the router:
+
+```yaml title="router.yaml"
+preview_response_cache:
+  enabled: true
+  subgraph:
+    all:
+      enabled: true
+      redis:
+        urls: ["redis://..."]
+        namespace: response_cache
+```
+
+### Required to start
+
+When active, the `required_to_start` option prevents the router from starting if it cannot connect to Redis. By default, the router starts without a connection to Redis, which would result in only using response caching sending the requests to subgraphs undisturbed:
+
+```yaml title="router.yaml"
+preview_response_cache:
+  enabled: true
+  subgraph:
+    all:
+      enabled: true
+      redis:
+        urls: ["redis://..."]
+        required_to_start: true
+```
+
+### Connection pool size
+
+The `pool_size` option defines the number of Redis connections the router opens. By default, the router opens five connections. If you have high traffic between the router and Redis, or if you observe latency in those requests, increase the pool size to reduce that latency. You can measure whether you need to increase this number by monitoring the `apollo.router.operations.response_cache.insert` and `apollo.router.operations.response_cache.fetch` [metrics](/router/performance/caching/response-caching/observability#fetchinsert). If the time to insert and fetch data from the cache is long, it might be because of delays in acquiring a connection from the pool:
```

Comment:
```suggestion
The `pool_size` option defines the number of Redis connections the router opens. By default, the router opens five connections. If you have high traffic between the router and Redis, or if you observe latency in those requests, increase the pool size to reduce that latency. You can measure whether you need to increase this number by monitoring the `apollo.router.operations.response_cache.insert` and `apollo.router.operations.response_cache.fetch` [response cache metrics](/router/performance/caching/response-caching/observability#fetchinsert). If the time to insert and fetch data from the cache is long, it might be because of delays in acquiring a connection from the pool:
```